service/src/test/java/uk/nhs/adaptors/gp2gp/common/configuration/AppInitializerTest.java

```diff
@@ -0,0 +1,40 @@
+package uk.nhs.adaptors.gp2gp.common.configuration;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.services.s3.S3Client;
+import uk.nhs.adaptors.gp2gp.common.storage.StorageConnectorConfiguration;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class AppInitializerTest {
+
+    private AppInitializer appInitializer;
+    private StorageConnectorConfiguration storageConnectorConfiguration;
+
+    @BeforeEach
+    void setUp() {
+        storageConnectorConfiguration = new StorageConnectorConfiguration();
+    }
+
+    @Test
+    void getNullWhenTrustStoreUrlDoesNotExists() {
+
+        storageConnectorConfiguration.setTrustStoreUrl(null);
+        appInitializer = new AppInitializer(storageConnectorConfiguration);
+
+        S3Client s3Client = appInitializer.getS3Client();
+
+        assertNull(s3Client);
+    }
+
+    @Test
+    void getNullWhenTrustStoreUrlDoesNotStartWithS3Prefix() {
+
+        storageConnectorConfiguration.setTrustStoreUrl("http://localhost");
+        appInitializer = new AppInitializer(storageConnectorConfiguration);
+
+        S3Client s3Client = appInitializer.getS3Client();
+
+        assertNull(s3Client);
+    }
```

Comment:
Thinking about this I don't think this behaviour is actually useful. The CustomTrustStore needs an S3Client to work, so having a CustomTrustStore created but without an S3Client is buggy.

Reply:
But in turn the creation of s3Client is depended on the trustStoreUrl and if that's not provided then s3Client also doesn't seem to be useful for the trustStoreEven if constructed... Looks like I will have to change it and raise an exception instead of returning null